packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -76,10 +98,7 @@ export class AwsS3Publish implements PublisherBase {
       ...(credentials &&
         accessKeyId &&
         secretAccessKey && {
-          credentials: {
-            accessKeyId,
-            secretAccessKey,
-          },
+          credentials: awsCredentials,
```

Comment:
Since `accessKeyId` and `secretAccessKey` will remain `undefined` in the case block above where `roleArn` is defined, `awsCredentials` won't be used here because of the `&&` chaining.

Perhaps it makes sense to do

```ts
    const storageClient = new aws.S3({
      ...(credentials &&
        awsCredentials && {
          credentials: awsCredentials,
```

Reply:
we could just make it this, no?
```ts
const storageClient = new aws.S3({
      credentials,
      ...(region && {
        region,
      }),
    });
```
 If credentials is undefined, it'll fallback to default behavior

Reply:
Yep - much cleaner.